### PR TITLE
Fix crash when calculating density of sum of normal and normal to the 5:th power

### LINKED ISSUE #24915: Crash when calculating density of sum of normal and normal to the 5:th power
Calculating the density of the sum of the fifth power of a normal distributed random variable and the random variable raises. Lower powers seems to work, but not higher. I think, although I am not sure, that the calculations are correct. It is just that the calculation at some point leads to a `ConditionSet` perhaps via a `Piecewise`.

```python

import sympy.stats as stats

# Crashes
X = stats.Normal('X', 0, sympy.Symbol("sigma")**2)
stats.density(X**5 + X)

# Works
X = stats.Normal('X', 0, sympy.Symbol("sigma")**2)
stats.density(X**3 + X)
```

```
Traceback (most recent call last):

  File "/tmp/ipykernel_17323/2537738759.py", line 1, in <module>
    stats.density(X**5 + X)

  File "/usr/lib/python3/dist-packages/sympy/stats/rv.py", line 922, in density
    return Density(expr, condition).doit(evaluate=evaluate, **kwargs)

  File "/usr/lib/python3/dist-packages/sympy/stats/rv.py", line 869, in doit
    result = pspace(expr).compute_density(expr, **kwargs)

  File "/usr/lib/python3/dist-packages/sympy/stats/crv.py", line 529, in compute_density
    fy = sum(fx(g) * abs(g.diff(y)) for g in gs)

TypeError: 'ConditionSet' object is not iterable
```

### PR DESCRIPTION
**Pull request by zu0feng**
#### References to other Issues or PRs

Fixes #24915.

#### Other comments
This is our group's pull request about the issue 24915, and this problem was solved by referring to issue #25037, because this is our group's first formal PR. If there is anything incorrect, please kindly understand and point it out.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by wermos** [comment]
Hi @zu0feng, it would be helpful if you could change the PR title to something more descriptive. You can refer to [this](https://docs.sympy.org/dev/contributing/new-contributors-guide/workflow-process.html#writing-pull-request-title-and-description) to see why we prefer descriptive PR titles 😄

**Comment by hanspi42** [comment]
Also, I advise to run some tests locally before you check it in. In your case, you could `cd` to where you have your local git copy of sympy and then

```
python bin/test quality 
python bin/doctest stats
python bin/test --slow stats
python bin/test stats
```

It would also be good if you add a test that will fail in case someone reverts this correction and  #24915 re-occurs.

File: sympy/stats/tests/test_compound_rv.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 from sympy.core.numbers import (oo, pi)
 from sympy.core.relational import Eq
 from sympy.core.singleton import S
-from sympy.core.symbol import symbols
+from sympy.core.symbol import symbols,Symbol
```
Comment (smichr) [inline sympy/stats/tests/test_compound_rv.py:5]:
```suggestion
from sympy.core.symbol import symbols, Symbol
```

File: sympy/stats/tests/test_compound_rv.py [hunk 2]
```diff
@@ -137,7 +137,10 @@ def test_Compound_Distribution():
     assert C.pdf(z, evaluate=True).simplify() == t**z*(t + 1)**(-k - z)*gamma(k \
                     + z)/(gamma(k)*gamma(z + 1))
 
-
+def test_issue_24915():
+    sigma = Symbol("sigma")
+    X = Normal('X', 0, sigma**2)
+    raises(ValueError, lambda: density(X**5 + X))
```
Comment (smichr) [inline sympy/stats/tests/test_compound_rv.py:143]:
```suggestion
    raises(ValueError, lambda: density(X**5 + X))


```

File: sympy/stats/crv.py [hunk 3]
```diff
@@ -531,11 +531,13 @@ def compute_density(self, expr, **kwargs):
 
         gs = solveset(expr - y, self.value, S.Reals)
 
-        if isinstance(gs, Intersection) and S.Reals in gs.args:
-            gs = list(gs.args[1])
+        
+        if isinstance(gs, Intersection):
+            if len(gs.args) == 2 and gs.args[0] is S.Reals:
+                        gs = gs.args[1]
```
Comment (smichr) [inline sympy/stats/crv.py:537]:
```suggestion
                gs = gs.args[1]
```

Comment (smichr) [reply sympy/stats/crv.py:537]: Can the intersection ever be longer than 2? Will the Reals always be in slot 1? Perhaps better to raise an error if the conditions you handle are not met, e.g. `else: raise NotImplementedError('anticipating Intersection(Reals, Set)')`.

**Comment by oscarbenjamin** [comment]
Superseded by gh-26095

**Pull request closed without merging**